source/Calamari/Kubernetes/ResourceStatus/ResourceRetriever.cs

```diff
@@ -56,13 +62,42 @@ private IEnumerable<Resource> GetChildrenResources(Resource parentResource, IKub
             }
 
             var result = kubectlGet.AllResources(childKind, parentResource.Namespace, kubectl);
-            var resources = ResourceFactory.FromListJson(result, options);
+
+            var resources = TryParse(ResourceFactory.FromListJson, result, options);
             return resources.Where(resource => resource.OwnerUids.Contains(parentResource.Uid))
-                .Select(child =>
-                {
-                    child.UpdateChildren(GetChildrenResources(child, kubectl, options));
-                    return child;
-                }).ToList();
+                            .Select(child =>
+                            {
+                                child.UpdateChildren(GetChildrenResources(child, kubectl, options));
+                                return child;
+                            }).ToList();
+        }
+
+        T TryParse<T>(Func<string, Options, T> function, string jsonString, Options options)
+        {
+            try
+            {
+                return function(jsonString, options);
+            }
+            catch (JsonException)
+            {
+                LogJsonStringError(jsonString, options);
+                throw;
+            }
```

Comment:
Is there any concern that the `JsonException` has any sensitive information in it? Just thinking for the case when the exception bubbles up and something else logs it out

Reply:
It doesn't log much of anything to be honest which is part of the reason why we've needed to add this. It does log line and column numbers so you can identify where it's failing to parse as well as a character or two which it doesn't expect, but it doesn't seem to log more than that.